src/foam/dao/index/TreeIndex.java

```diff
@@ -115,6 +116,27 @@ public Object put(Object state, FObject value) {
       prop_, key, value, tail_);
   }
 
+  @Override
+  public Object update(Object state, FObject oldValue, FObject newValue) {
+
+      Object key;
+      Object oldKey;
+      try {
+        key = prop_.f(newValue);
+        oldKey = prop_.f(oldValue);
+      } catch (ClassCastException e) {
+        return state;
+      }
+      if ( oldKey != null && key.equals(oldKey) )
```

Comment:
Why is oldKye != null needed?

Reply:
tests was failing. so my guess was that what's happen  if property value is null